### ISSUE TITLE
Pods with hyphens in the header_dir can break between versions 1.6 & 1.7

### ISSUE THREAD

**Issue by jdee** [CocoaPods/CocoaPods#8857]
Any pod with a header_dir that includes hyphens can have broken imports when updating to CocoaPods 1.7. This may affect numerous React Native packages.

* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

After updating to CocoaPods 1.7, I ran `pod update` in a number of apps.

## What did you expect to happen?

I expected everything to build cleanly.

## What happened instead?

Because my SDK module name includes hyphens (react-native-branch), CocoaPods changed the location where this is installed to react_native_branch. So CocoaPods < 1.7 requires `#import <react-native-branch/RNBranch.h>`, and CP >= 1.7 requires `#import <react_native_branch/RNBranch.h>`.

### Stack

```
   CocoaPods : 1.7.0
        Ruby : ruby 2.6.3p62 (2019-04-16 revision 67580) [x86_64-darwin18]
    RubyGems : 3.0.3
        Host : Mac OS X 10.14.5 (18F132)
       Xcode : 10.2.1 (10E1001)
         Git : git version 2.20.1 (Apple Git-117)
Ruby lib dir : /Users/jdee/.rvm/rubies/ruby-2.6.3/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ a5f13772a78819a9c52433b530ef75c77ed51944
```

### Installation Source

```
Executable Path: /Users/jdee/.rvm/gems/ruby-2.6.3/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.4
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.1.0
cocoapods-trunk       : 1.3.1
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
platform :ios, '9.0'

target 'testbed_simple' do
  pod 'React',
    path: '../node_modules/react-native',
    subspecs: %w[
      Core
      CxxBridge
      DevSupport
      RCTAnimation
      RCTImage
      RCTText
      RCTNetwork
    ]
  pod 'yoga', path: '../node_modules/react-native/ReactCommon/yoga'
  pod 'Folly', podspec: '../node_modules/react-native/third-party-podspecs/Folly.podspec'
  pod 'react-native-branch', :path => '../node_modules/react-native-branch'

end
```

## Project that demonstrates the issue

See https://github.com/BranchMetrics/react-native-branch-deep-linking-attribution/tree/master/examples/testbed_simple. Note this is a React Native app. You must first run `yarn` in that directory and then cd to the ios subdirectory to find the Podfile. For now I've pinned to 1.6 in my Gemfile (at the repo root). You can update the Gemfile and try `bundle exec pod update` in the testbed_simple/ios directory to see where it breaks the import in the AppDelegate.

**Comment by paulb777** [comment]
Looks like a dupe of #8836 which is planned to be fixed in 1.7.1

**Comment by jdee** [comment]
@paulb777 Thanks. I don't know why I didn't find that. We'll all look forward to 1.7.1.
